cmd/object-handlers.go

```diff
@@ -1541,6 +1550,9 @@ func (api objectAPIHandlers) PutObjectPartHandler(w http.ResponseWriter, r *http
 		}
 	}
 
+	if globalAutoEncryption && !crypto.SSEC.IsRequested(r.Header) {
+		r.Header.Add(crypto.SSEHeader, crypto.SSEAlgorithmAES256)
+	}
```

Comment:
SSE-S3 header is not required for UploadPart request per S3 spec. This can be removed

Reply:
btw, CopyObjectPart needs to add sse-s3 headers on destination side if autoEncryption is on.

Reply:
Oh, correct, done. Also added check to CopyObjectPart